server/services/userService.ts

```diff
@@ -30,7 +33,28 @@ export default class UserService {
     }
   }
 
-  getUserCaseLoads(token: string): Promise<CaseLoad[]> {
-    return this.prisonApiClientBuilder(token).getUserCaseLoads()
+  private async ensureConnected() {
+    if (!this.redisClient.isOpen) {
+      await this.redisClient.connect()
+    }
+  }
+
+  async getUserCaseLoads(token: string, username: string): Promise<CaseLoad[]> {
+    try {
+      await this.ensureConnected()
+      const redisData = await this.redisClient.get(username)
+      if (redisData) {
+        return JSON.parse(redisData).caseLoads as CaseLoad[]
+      }
+
+      const caseLoads = await this.prisonApiClientBuilder(token).getUserCaseLoads()
+      if (caseLoads.length <= 9) {
```

Comment:
don't really understand why it is a `9` here - are you relying on each caseload response having at least 5 entries?  Couldn't you use the size of the array instead?

Reply:
Ha, good spot! That was for testing as my user has more than 1

Reply:
As caught by my int test